### ISSUE TITLE
 how to set up a proxy?

### ISSUE THREAD

**Issue by waitandsoon** [th-ch/youtube-music#189]
The application is excellent! 
but there is a problem: how to set up a proxy? 
Because in some "magic" countries it is not possible to use youtube normally.

**Comment by Bellisario** [comment]
I'm not very expert about proxies, but I think that you can use a normal proxy application (for browsers), because YouTube Music App, it's based by Electron (which in turn it's based by Chromium).

**Comment by waitandsoon** [comment]
I found an effective method, share it and put it here. Maybe it's useful to anyone.

**Mac platform：**
```
cd "/Applications/YouTube Music.app/Contents/MacOS/"
sudo mv "YouTube Music" abc.real
sudo printf '#!/bin/bash\ncd "/Applications/YouTube Music.app/Contents/MacOS"\n"/Applications/YouTube Music.app/Contents/MacOS/abc.real"
-proxy-server=socks5://127.0.0.1:9999
"$@"\n' > YouTube Music

```
**Windows platform：**
On the desktop shortcut, add parameters：

`--proxy-server=socks5://127.0.0.1:9999`

**Comment by th-ch** [comment]
Hey @waitandsoon, thanks for the feedback! https://github.com/th-ch/youtube-music/commit/61e7124516571a63c7155937e827ba95dc34b9cb adds support for proxy in advanced options (in the menu, `Options` > `Advanced options`):
```json
"options": {
    …
    "proxy": "socks5://127.0.0.1:9999"
},
…
```
Then restarting the app should work! This change will land in the next version of the app.

**Comment by waitandsoon** [comment]
> Hey @waitandsoon, thanks for the feedback! [61e7124](https://github.com/th-ch/youtube-music/commit/61e7124516571a63c7155937e827ba95dc34b9cb) adds support for proxy in advanced options (in the menu, `Options` > `Advanced options`):
> 
> ```json
> "options": {
>     …
>     "proxy": "socks5://127.0.0.1:9999"
> },
> …
> ```
> 
> Then restarting the app should work! This change will land in the next version of the app.

Thank you very much, I will try it right away

**Comment by miyukowo** [comment]
Hello, I need help. I tried doing the same thing and it doesn't work, any answers on this?
![image](https://user-images.githubusercontent.com/40559090/117036265-713fb200-ad2f-11eb-9b71-c9f11cff6a72.png)
![image](https://user-images.githubusercontent.com/40559090/117036290-78ff5680-ad2f-11eb-8f2a-e2ebfc86bd9c.png)

**Comment by Araxeus** [comment]
@miyukowo This is currently only available in the git branch
so if you want to use it right now, you have to build the app yourself.

hopefully a new packaged version will be released soon than will include this and many more new features

**Comment by miyukowo** [comment]
@Araxeus Ah, gotcha. Appreciate the speedy reply.

**Comment by Araxeus** [comment]
@waitandsoon @miyukowo Hey, I tried to make a proxy check (check if proxy address is valid on launch)
could you try and see if it launches properly for you with a **valid proxy address**?

#### To test - clone this branch https://github.com/Araxeus/youtube-music/tree/check-proxy

> Its important that you test first without those changes that you can connect using your valid proxy address

**Comment by miyukowo** [comment]
Hello @Araxeus,
It would be great if you can create and upload windows binaries as I really don't know how to build the app itself.
Sorry for the trouble and thank you for the assist!

**Comment by Araxeus** [comment]
@miyukowo
The errors you got only means you need to install node version 12, and then you should be able to run `yarn build:win`

But anyways, I created and uploaded the binaries:
https://drive.google.com/drive/folders/1Kfa4USQtoUjZTZLzgToDM6kMELdkRqgC?usp=sharing

**Comment by miyukowo** [comment]
@Araxeus 
Hello, I've requested access for the binaries file, can you please approve?

**Comment by Araxeus** [comment]
Sorry I've just updated the folder settings, you should now be able to access it

**Comment by miyukowo** [comment]
Thank you, already downloading it. I'll be testing the proxy check in a bit.

**Comment by miyukowo** [comment]
Hello, I'm back after messing around with the build you made.
Good news, the proxy check worked!

![image](https://user-images.githubusercontent.com/40559090/118365260-24c56380-b5c6-11eb-984e-2bd02bd0bdb9.png)

Bad news - I've tried a few socks5/https on the internet but sadly, none worked. Any suggestions?

**Comment by Araxeus** [comment]
When you say "the proxy check worked" you mean it let you through with a proxy bound to a local address?

Or you just meant it always show an error message? 😆

**Comment by miyukowo** [comment]
Yes, the latter one. Every single proxies I tried doesn't work, so that might have something to do with either the program or the proxies. But if I leave it empty it does work.

![image](https://user-images.githubusercontent.com/40559090/118365827-7969de00-b5c8-11eb-8aac-a7cf1ed98d7b.png)

Thank you for the help.

**Comment by Araxeus** [comment]
Ok actually I think I'm abandoning this "proxy-check" feature.

Of course you will still be able specify a proxy (soon in menu -> options -> advanced -> proxy)

If you specify a bad proxy - you can actually just change it easily in the menu and restart the app

**Comment by miyukowo** [comment]
Good idea, let's settle on that then.
